### PR TITLE
Sprint4

### PR DESCRIPTION
**Pull request by linabenefit**
В этот раз работа шла очень тяжело:(
Бургер вроде получился.
Много времени потеряла на галерее Fancybox. Она никак не работала, пока я не откатилась на предыдущую версию Fancybox.

После этого я намудрила с каруселью Swiper -  расстояния между картинками оставляет большие, а когда я добавляю стрелки - они вообще переносятся под header. Перед отправкой убрала с них классы навигации стрелок. С каруселью пока грустно, не справилась с ней. 

Плавный скролл до якорных элементов тоже получилось сделать.

### CODE REVIEW

File: src/normalize.css
Comment (euge-huge) [inline src/normalize.css:1]: В package.json подключен пакет "normalize.css": "^8.0.1", 

В scss его подключить не удалось с помощью @use?

File: src/scss/media.scss [hunk 1]
```diff
@@ -36,21 +45,408 @@
...
+        display: none;
+    }
+
+
+
+    
+
+
+      
+
```
Comment (euge-huge) [inline src/scss/media.scss:221]: Очень много пустых строк в файлах, очень необходимо форматирование, стандартных средств VS Code в виде Shift+Alt+F должно быть достаточно

File: src/scss/media.scss
Comment (euge-huge) [inline src/scss/media.scss:1]: Адаптивная верстка сильно улучшилась, молодец!
Но остаются моменты, которые портят верстку:
image
Полоса:
Screenshot 2023-10-24 at 11 12 36

File: src/scss/media.scss [hunk 2]
```diff
@@ -36,21 +45,408 @@
...
+    .body.lock {
+        overflow: hidden;
+    }
+
+    /* BURGER */
+
+  .header-top {
+   
+   max-width: 768px;
```
Comment (euge-huge) [inline src/scss/media.scss:83]: max-width: 100% не работает?
От абсолютных размеров нужно уходить

File: src/scss/media.scss [hunk 3]
```diff
@@ -36,21 +45,408 @@
...
+    }
+
+    .header__burger.active::before {
+        transform: rotate(45deg);
+        top: 9px;
+        
+    }
+    .header__burger.active span {
+        transform: scale(0);
+    }
```
Comment (euge-huge) [inline src/scss/media.scss:156]: Почему бы не воспользоваться вложенностью?
.header {
  &__burger {
    &::before {}
  }
}

Так код приобретает совсем другой вид

File: src/scss/media.scss [hunk 4]
```diff
@@ -36,21 +45,408 @@
...
+   .paragraph-second {
+    width: 450px;
+   }
+
+   .about-columns {
+    margin-top: 400px;
+   }
+
+   .column-icons {
+    margin: 10%;
```
Comment (euge-huge) [inline src/scss/media.scss:321]: Отступы в процентах не всегда удачно получается использовать

**Comment by euge-huge** [comment]
В gulpfile нужно добавить импорты 
```
const webpackStream = require('webpack-stream');
 const rename = require('gulp-rename');
```

И добавить новую таску

```
 function buildJS() {
     return src('src/js/index.js')
     .pipe(webpackStream(require('./webpack.config')))
     .pipe(rename('main.min.js'))
     .pipe(dest('src/js'))
     .pipe(dest('dist/js'))
     .pipe(browserSync.stream());
 }
```
 
Добавить к серверу и билду
```
 function serve() {
     watch(['src/js/**/*.js', '!src/js/**/*.min.js'], buildJS);
     watch('src/scss/**/*.scss', buildSass);
     watch('src/**/*.html', buildHtml);
 }

function createDevServer() {
     browserSync.init({
         server: 'src/',
         notify: false,
     })
 }

 exports.build = series(cleanDist, parallel(buildSass, buildHtml, copy, buildJS));
 exports.default = series([buildSass, buildJS], parallel(createDevServer, serve));
```

File: src/scss/style.scss [hunk 5]
```diff
@@ -453,6 +487,242 @@ a:hover {
...
+  border-style: solid;
+  margin-bottom: 20px;
+}
+
+.telephone {
+  width: 375px;
+  height: 46px;
+  border-color: #F8F6F7;
+  border-radius: 8px;
+  border-style: solid;
```
Comment (euge-huge) [inline src/scss/style.scss:708]: Я так думаю среди этих классов есть элементы по БЭМ

File: src/scss/style.scss [hunk 6]
```diff
@@ -453,6 +487,242 @@ a:hover {
...
+.record-describe {
+  font-size: 14px;
+  font-weight: 400;
+  width: 374px;
+  margin-bottom: 35px;
+}
+
+.username {
+  width: 375px;
+  height: 46px;
```
Comment (euge-huge) [inline src/scss/style.scss:696]: Все эти размеры как раз и мешают сделать хороший адаптив

Comment (euge-huge) [reply src/scss/style.scss:696]: И вдругих местах в том числе

Comment (euge-huge) [reply src/scss/style.scss:696]: Размеры задаем только блокам, которые должны быть при любых размерах экрана одного размера: некоторые картинки, карточки и т.д)

File: src/index.html [hunk 7]
```diff
@@ -154,23 +175,25 @@ <h3 class="card__heading">Парафинотерапия</h3>
...
-            <li class="navigation-item"><a href="#" class="navigation__name">Маникюр и педикюр</a></li>
-            <li class="navigation-item"><a href="#" class="navigation__name">Макияж</a></li>
-            <li class="navigation-item"><a href="#" class="navigation__name">Брови и ресницы</a></li>
-            <li class="navigation-item"><a href="#" class="navigation__name">Массаж</a></li>
+            <li class="navigation__tab"><a href="#tab01" class="navigation__name">Стрижка и укладка</a></li>
+            <li class="navigation__tab"><a href="#tab02" class="navigation__name">Косметология</a></li>
+            <li class="navigation__tab"><a href="#tab03" class="navigation__name">Маникюр и педикюр</a></li>
+            <li class="navigation__tab"><a href="#tab04" class="navigation__name">Макияж</a></li>
+            <li class="navigation__tab"><a href="#tab05" class="navigation__name">Брови и ресницы</a></li>
+            <li class="navigation__tab"><a href="#tab06" class="navigation__name">Массаж</a></li>
```
Comment (euge-huge) [inline src/index.html:191]: Нет большого смысла добавлять ссылку на табы, это актуально в шапке для перемещения, в этом же случае лучше использовать JS только

File: src/index.html [hunk 8]
```diff
@@ -154,23 +175,25 @@ <h3 class="card__heading">Парафинотерапия</h3>
...
         </ul>
     </nav>
 
 
-<div class="prices__costs">
+<div id="tab01" class="prices__costs">
```
Comment (euge-huge) [inline src/index.html:196]: id задавать лучше с понятными человеку названиями, так проще ориентироваться)

+ пока все табы пропали, нужно разбираться

File: src/index.html [hunk 9]
```diff
@@ -349,4 +941,14 @@ <h1 class="masters-title">Мастера</h1>
...
 </div>
 </div>
-</footer>
\ No newline at end of file
+</footer>
+<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.7.1/jquery.min.js"></script>
+<script src="https://cdn.jsdelivr.net/npm/@fancyapps/ui@4.0/dist/fancybox.umd.js"></script>
+<script src="https://cdn.jsdelivr.net/npm/swiper@10/swiper-bundle.min.js"></script>
+
+<script src="./js/index.js"></script>
```
Comment (euge-huge) [inline src/index.html:949]: Файл подключается при локальном запуске сервера (start), но при билде не подключается (поправить gulpfile)

File: src/index.html
Comment (euge-huge) [inline src/index.html:1]: `отлично`

Сам код для верстки выглядит хорошо!

File: src/js/index.js [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+ 
+    });
+});
+
+$(document).ready(function() {
+    $('.navigation-item').click(function(event) {
+        $('.header__burger,.navigation').removeClass('active');
+        $('body').removeClass('lock')
+    });
+});
```
Comment (euge-huge) [inline src/js/index.js:18]: `отлично`

С бургером все неплохо, похоже, получилось

File: src/js/index.js [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+
+$('a[href^="#"]').on('click', function() {
+
+    let href = $(this).attr('href');
+
+    $('html, body').animate({
+        scrollTop: $(href).offset().top
+    });
+    return false;
+});
```
Comment (euge-huge) [inline src/js/index.js:65]: Тоже неплохо, универсально, код будет работать для всех ссылок с href для id на странице

File: src/js/index.js [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+    const Username = form.querySelector('[name="Username"]'), 
+    Usertel = form.querySelector('[name="Usertel"]');
+
+    const values = {
+        Username: Username.value,
+        Usertel: Usertel.value
+    };
+
+    console.log ('Получение данных',values);
+}
```
Comment (euge-huge) [inline src/js/index.js:37]: Функция есть, но никак не используется, поэтому в консоль ничего не выводится)

 Нужно добавить слушатель на отправку формы:
`form.addEventListener("submit", retrieveFormValue);
`

Также не мешало бы все обернуть в $(document).ready()

Можно обернуть сразу все в данном файле, чтобы не писать каждый раз

File: src/js/index.js [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+new Swiper('.image-slider'), {
+    
+    navigation: {
+    nextEl:'swiper-button-next',
+    prevEl:'swiper-button-prev'
+    },
+
+    slidesPerView: 4,
+    
+}
```
Comment (euge-huge) [inline src/js/index.js:52]: Со слайдером очень тяжело помочь, пока верстка не валидна (нужно срочно пройтись по всему html и найти лишние теги, либо закрывающие лишние, но с html однозначно есть проблемы сейчас), после починки верстки уже можно решать проблему со слайдером, но проблемы однозначно пока в стилях, конфликт нужно будет решать после предыдущих шагов, когда верстка встанет нормально

File: src/index.html
Comment (euge-huge) [reply src/index.html:1]: Но есть большая проблема, которую нужно срочно решить, я о ней писал в предыдущих спринтах несколько раз

Где-то есть лишние теги, которые делают верстку невалидной, с ней очень тяжело работать, все становится непредсказуемым
Я говорю о моментах вида (лишний </div> ломает валидность верстки)
```
<div class="container">
  ...
</div>
</div>
```

Эту проблему было проще всего решить, когда кода было мало, теперь нужно по кусочкам аккуратно разобрать весь код и решить ее, это займет время, но без этого не рекомендую продолжать работу

**Review by euge-huge — APPROVED** [review]
Работу зачел!

Но есть достаточно проблем, которые требуют срочного решения, оставил все в комментариях.

Самая главная проблема, из-за которой, возможно, появляются все другие - не валидность верстки
Prettier подскажет, когда все будет починено, внизу при нажатии Shift+Alt+F перестанет гореть красная копка
Screenshot 2023-10-24 at 11 55 10
